dvc/repo/__init__.py

```diff
@@ -202,6 +202,18 @@ def __init__(
     def __str__(self):
         return self.url or self.root_dir
 
+    @staticmethod
+    @contextmanager
+    def open(url, *args, **kwargs):
+        if os.path.isdir(url):
+            yield Repo(url, *args, **kwargs)
+            return
+
+        from dvc.external_repo import external_repo
+
+        with external_repo(url, *args, **kwargs) as repo:
```

Comment:
What this really is is a `Repo.clone` with a temporary dir. Having `Repo.open()` is very convenient when you just want to access the repo, no matter local or remote (in which case temp clone is totally fine).